lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +204,39 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self.config = config
+        self.urls = []
+        self._seed_urls = config.get_seed_urls()
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        all_links_bs = article_bs.find_all('a')
+        for links_bs in all_links_bs:
+            link = links_bs.get('href')
+        # link = article_bs.get('href')
+            if link is not None and link.count('/') == 6 and '/cat/politroom/' in link and link[:8] == 'https://':
+                self.urls.append(link)
+        return ''
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        for seed_url in self._seed_urls:
+            try:
+                response = make_request(seed_url, self.config)
+                main_bs = BeautifulSoup(response.text, 'lxml')
+                self._extract_url(main_bs)
+            except WebsiteError:
```

Comment:
it is cool that you've thought about implementing another exception, but I don't see where it can be risen. If you wanted to throw it from the `make_request` function, consider also including information on the response code, URL etc.

Reply:
I rewrote the code, but incomprehensible errors began to occur related to files where not my code was written, but the code of checks. To be honest, it's hard to understand where I specifically made a mistake